type_description_interfaces/msg/FieldType.msg

```diff
@@ -0,0 +1,157 @@
+uint8 FIELD_TYPE_UINT16_BOUNDED_SEQUENCE = 101
+uint8 FIELD_TYPE_INT32_BOUNDED_SEQUENCE = 102
+uint8 FIELD_TYPE_UINT32_BOUNDED_SEQUENCE = 103
+uint8 FIELD_TYPE_INT64_BOUNDED_SEQUENCE = 104
+uint8 FIELD_TYPE_UINT64_BOUNDED_SEQUENCE = 105
+uint8 FIELD_TYPE_FLOAT_BOUNDED_SEQUENCE = 106
+uint8 FIELD_TYPE_DOUBLE_BOUNDED_SEQUENCE = 107
+uint8 FIELD_TYPE_LONG_DOUBLE_BOUNDED_SEQUENCE = 108
+uint8 FIELD_TYPE_CHAR_BOUNDED_SEQUENCE = 109
+uint8 FIELD_TYPE_WCHAR_BOUNDED_SEQUENCE = 110
+uint8 FIELD_TYPE_BOOLEAN_BOUNDED_SEQUENCE = 111
+uint8 FIELD_TYPE_BYTE_BOUNDED_SEQUENCE = 112
+uint8 FIELD_TYPE_STRING_BOUNDED_SEQUENCE = 113
+uint8 FIELD_TYPE_WSTRING_BOUNDED_SEQUENCE = 114
+uint8 FIELD_TYPE_FIXED_STRING_BOUNDED_SEQUENCE = 115
+uint8 FIELD_TYPE_FIXED_WSTRING_BOUNDED_SEQUENCE = 116
+uint8 FIELD_TYPE_BOUNDED_STRING_BOUNDED_SEQUENCE = 117
+uint8 FIELD_TYPE_BOUNDED_WSTRING_BOUNDED_SEQUENCE = 118
+
+# Unbounded Sequence Types
+uint8 FIELD_TYPE_NESTED_TYPE_UNBOUNDED_SEQUENCE = 145
+uint8 FIELD_TYPE_INT8_UNBOUNDED_SEQUENCE = 146
+uint8 FIELD_TYPE_UINT8_UNBOUNDED_SEQUENCE = 147
+uint8 FIELD_TYPE_INT16_UNBOUNDED_SEQUENCE = 148
+uint8 FIELD_TYPE_UINT16_UNBOUNDED_SEQUENCE = 149
+uint8 FIELD_TYPE_INT32_UNBOUNDED_SEQUENCE = 150
+uint8 FIELD_TYPE_UINT32_UNBOUNDED_SEQUENCE = 151
+uint8 FIELD_TYPE_INT64_UNBOUNDED_SEQUENCE = 152
+uint8 FIELD_TYPE_UINT64_UNBOUNDED_SEQUENCE = 153
+uint8 FIELD_TYPE_FLOAT_UNBOUNDED_SEQUENCE = 154
+uint8 FIELD_TYPE_DOUBLE_UNBOUNDED_SEQUENCE = 155
+uint8 FIELD_TYPE_LONG_DOUBLE_UNBOUNDED_SEQUENCE = 156
+uint8 FIELD_TYPE_CHAR_UNBOUNDED_SEQUENCE = 157
+uint8 FIELD_TYPE_WCHAR_UNBOUNDED_SEQUENCE = 158
+uint8 FIELD_TYPE_BOOLEAN_UNBOUNDED_SEQUENCE = 159
+uint8 FIELD_TYPE_BYTE_UNBOUNDED_SEQUENCE = 160
+uint8 FIELD_TYPE_STRING_UNBOUNDED_SEQUENCE = 161
+uint8 FIELD_TYPE_WSTRING_UNBOUNDED_SEQUENCE = 162
+uint8 FIELD_TYPE_FIXED_STRING_UNBOUNDED_SEQUENCE = 163
+uint8 FIELD_TYPE_FIXED_WSTRING_UNBOUNDED_SEQUENCE = 164
+uint8 FIELD_TYPE_BOUNDED_STRING_UNBOUNDED_SEQUENCE = 165
+uint8 FIELD_TYPE_BOUNDED_WSTRING_UNBOUNDED_SEQUENCE = 166
+
+# Identifying number for the type of the field, using one of the above constants.
+uint8 type_id
+
+# Only used when the type is an array or a bounded sequence.
+# In the case of an array, this is the fixed length of the array.
+# In the case of a bounded sequence, this is the maximum size of the sequence.
+# In all other cases this field is unused.
+uint64 length
+
+# Only used when the type is a fixed or bounded string/wstring, or a array/sequence of those.
+# In the case of a fixed string/wstring, it is the fixed length of the string.
+# In the case of a bounded string/wstring, it is the maximum length of the string.
+# In the case of an array/sequence of fixed string/wstring, it is the fixed length of the strings.
+# In the case of an array/sequence of bounded string/wstring, it is the maximum length of the strings.
+# It is not currently possible to have different string lengths per element in the array/sequences.
```

Comment:
@wjwwood I don't think this statement is true (e.g. `sequence<string>`, and `sequence<string, 255>`).

```suggestion
``` 

Neither of those cases are in the "type is a fixed or bounded string/wstring" category, so I think this statement can just be removed.

Reply:
I'm not sure I agree.  If we have a sequence of fixed string, for instance, this field represents the fixed length of each of the strings.  It is not possible with this structure to have a sequence of fixed strings, and have some of the strings be of length 10 and some of length 20.  So I think this comment still applies.  @allenh1 were you thinking about something else?

Reply:
What Chris said is what I was talking about with this line, specifically you cannot have heterogenous sequences and therefore cannot have a sequence with one string fixed to 10 and another fixed to 100 because those are different types.

Reply:
I see. I think we are operating on different definitions for the word "length".